src/details/ArborX_Callbacks.hpp

```diff
@@ -195,13 +199,17 @@ void check_valid_callback(Callback const &callback, Predicates const &)
         std::is_void<
             Kokkos::detected_t<Experimental_CallbackArchetypeExpression,
                                Callback, Predicate, int>>{})) ||
-          std::is_same<PredicateTag, NearestPredicateTag>{},
+          std::is_same<PredicateTag, NearestPredicateTag>{} ||
+          std::is_same<PredicateTag,
+                       Experimental::OrderedIntersectionPredicateTag>{},
```

Comment:
It does not make sense you are using `invoke_callback_and_check_early_exit` in the traversal.